core/node/state_keeper/src/io/mempool.rs

```diff
@@ -298,6 +298,31 @@ impl StateKeeperIO for MempoolIO {
                         .await?;
                     continue;
                 }
+
+                // Reject transactions that violate block.timestamp constraints. Such transactions should be
+                // rejected at the API level, but we need to protect ourselves in case if a transaction
+                // goes outside of the allowed range while being in the mempool
+                let now = SystemTime::now()
```

Comment:
Same question as for the VM tracer: why is time acquired here instead of using one from the latest block?